guidance/rust_code/src/graph/conversions.rs

```diff
@@ -0,0 +1,111 @@
+
+use super::{GPSPoint, TangentialPoint, Graph, Edge, Vertex};
+use nalgebra::{Vector3, Matrix3};
+
+// Distance from the earth relative to the equator (m)
+const RO: f64 = 6378137.0;
+
+// Ecentricity of the earth (unitless?)
+const E: f64 = 0.0818;
+
+// Front entrance of king
+const ORIGIN: GPSPoint = GPSPoint {
+    lat: 34.6147979,
+    long: -112.4509615,
+    height: 1582.3
+};
+
+pub trait IntoTangential {
+    type Output;
+
+    fn into_tangential(self) -> Self::Output;
+}
+
+impl IntoTangential for GPSPoint {
+
+    type Output = TangentialPoint;
+
+    fn into_tangential(self) -> TangentialPoint {
+
+        // vector from center of the earth to an arbitrary point on the map
+        let r_ek_e = self.to_rad().to_xyz();
+
+        // vector from the center of the earth to the origin of our tangential frame
+        let r_eb_e = ORIGIN.to_rad().to_xyz();
+
+        // Vector that goes from origin to an arbitrary point of the map.
+        // Still being expressed in terms of a rectangular frame with its origin at the center of
+        // the earth.
+        let r_kb_e = r_eb_e - r_ek_e;
+
+        // The transformation matrix that converts the basis or our coordinate system
+        let c_e_k = Matrix3::new(
+            -(self.to_rad().long.cos() * self.to_rad().lat.sin()),       -ORIGIN.long.sin(),   -(ORIGIN.lat.cos() * ORIGIN.long.cos()), 
+            -(ORIGIN.lat.sin() * ORIGIN.long.sin()),                      ORIGIN.long.cos(),   -(ORIGIN.lat.cos() * ORIGIN.long.sin()),
+              ORIGIN.lat.cos(),                                           0.0,                  -ORIGIN.lat.sin() 
+        ).try_inverse().unwrap();
+
+        let result = c_e_k * r_kb_e;
+
+        TangentialPoint{
+            x: result[0],
+            y: result[1],
+            z: result[2]
+        }
+    }
+}
+
+impl GPSPoint {
+
+    /// Converts spherical cords to rectangular
+    fn to_xyz(&self) -> Vector3<f64> {
+        
+        // The distance from the center of the earth relative to where you are (m)
+        let re = RO / (1.0 - self.lat.sin().powi(2)*E.powi(2)).sqrt();
+
+        Vector3::new(
+            (re + self.height) * self.lat.cos() * self.long.cos(),
+            (re + self.height) * self.lat.cos() * self.long.sin(),
+            ((re + re*E.powi(2)) + self.height) * self.lat.sin()
+        )
+    }
```

Comment:
Didn't exactly address my earlier comment. I would like this function to be a direct conversion from a GPS point as if it had been read directly from the map into the same point but in ECEF xyz coordinates. Requiring that there be a different function in the way (which as far as I can tell isn't being used for anything else) so that you have to remember to call point.to_rad().to_xyz() every time you call this function is allowing a possibility that you can forget and call point.to_xyz() without the compiler ever noticing. 

Either do the built-in to_radians() conversion in this function and get rid of the to_rad() function below or make a new type that is a GPSPointRad, distinct from the existing GPSPoint, so that at no two points in the code does the GPSPoint type either represent radians or degrees only to be inferred from context. 

I'll happily argue this more directly in person if you aren't convinced